src/equilibrate_utils.jl

```diff
@@ -0,0 +1,726 @@
+    foreach(constraint -> push!(func_expr.args[1].args,recursive_replace_vars!(balance_poly(constraint), rn)),MacroTools.striplines(constraints...).args)
+    return func_expr
+    output = Expr(:block)
+    for constraint in MacroTools.striplines(constraints...).args
+        push!(output.args,Expr(:escape,:(internal___add___constraint!($rn))))
+        push!(output.args[end].args[1].args,recursive_replace_vars!(balance_poly(constraint), rn))
+    end
+    return output
+end
+#Changes a polynomial expression of the form p(x) = q(x) to p(x) - q(x). Used by add_constraint(s).
+function balance_poly(poly::Expr)
+    (poly.head != :call) && (return :($(poly.args[1])-$(poly.args[2])))
+    return poly
+end
+#Complicated function which replaces the named varriables in the expression supplied to @fix_concentration with the corresponding polyvar stored in the reaction network.
+function recursive_replace_vars!(expr::Union{ExprValues,LineNumberNode}, rn::Symbol)
+    if (typeof(expr) == Symbol)&&(expr!=:+)&&(expr!=:-)&&(expr!=:*)&&(expr!=:^)&&(expr!=:/)
+        return :(in($(QuoteNode(expr)),$(rn).syms) ? $(rn).equilibrate_content.polyvars.x[$(rn).syms_to_ints[$(QuoteNode(expr))]] : $(rn).equilibrate_content.polyvars.p[$(rn).params_to_ints[$(QuoteNode(expr))]])
+    elseif typeof(expr) == Expr
+        foreach(i -> expr.args[i] = recursive_replace_vars!(expr.args[i], rn), 1:length(expr.args))
+    end
+    return expr
+end
+#Function which does the actual work of adding the fixed concentration information. Not meant to be called directly, but is called through the fixed_concentration macro.
+function internal___add___constraint!(rn::DiffEqBase.AbstractReactionNetwork, constraints::Polynomial...)
+    check_is_polynomial(rn)
+    foreach(constraint -> push_constraint!(rn,constraint),constraints)
+    remove_replicate_polys!(rn)
+end
+#Replicate polynomials might be bad for HomotopyContinuation, this function removes them when they might occur.
+function remove_replicate_polys!(rn::DiffEqBase.AbstractReactionNetwork)
+    for i = length(get_equi_poly(rn)):-1:2, j = (i-1):-1:1
+        (get_equi_poly(rn)[i]==get_equi_poly(rn)[j])&&deleteat!(get_equi_poly(rn),i)
+        (get_equi_poly(rn)[i]==-get_equi_poly(rn)[j])&&deleteat!(get_equi_poly(rn),i)
+    end
+end
+
+#--- Makes homotopy continuation templates to be used at a later stage when solving for fixed points ---#
+#Can be called separatly, but will otherwise be called first time a steady state is to be found. Solves the system once using random parameters. Saves the solution as a template to be used for further solving.
+"""
+    make_hc_template(reaction_network)
+
+    Makes a template for solving the riven reaction networks steady states using homotopy continuation. Generally not needed as this is genrated automatically when netowkr is first solved *if one does not already exists). However, it can be sued to replace the current template, for whatever reason.
```

Comment:
```suggestion
    Makes a template for solving for the given reaction networks steady states using homotopy continuation. Generally not needed as this is generated automatically when network is first solved *if one does not already exists*. However, it can be sued to replace the current template, for whatever reason.
```

Reply:
Fixed.